Tasks/AWSShellScript/TaskOperations.ts

```diff
@@ -27,20 +27,29 @@ export class TaskOperations {
 
             if (this.taskParameters.scriptType === inlineScriptType) {
                 const tempDir = SdkUtils.getTempLocation()
-                const fileName = `awsshellscript_${process.pid}.sh'`
+                const fileName = `awsshellscript_${process.pid}.sh`
                 scriptPath = path.join(tempDir, fileName)
-                tl.writeFile(scriptPath, this.taskParameters.inlineScript)
+                fs.writeFileSync(scriptPath, this.taskParameters.inlineScript, { encoding: 'utf-8' })
             } else {
                 scriptPath = this.taskParameters.filePath
             }
 
-            let cwd = this.taskParameters.cwd
-            if (!cwd && !this.taskParameters.disableAutoCwd) {
-                cwd = path.dirname(scriptPath)
+            let workingDirectory: string
+            if (this.taskParameters.disableAutoCwd) {
+                workingDirectory = this.taskParameters.workingDirectory
+            } else {
+                workingDirectory = path.dirname(scriptPath)
+            }
+
+            // The solution to this not working on windows comes from doing it like it's done in
+            // the ms task: https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/BashV3/bash.ts
+            // Thanks to the ms team opensourcing their tasks for the solution logic
+            if (process.platform === 'win32') {
+                scriptPath =
+                    (await this.translateWindowsPath(path.dirname(scriptPath))) + `/${path.basename(scriptPath)}`
```

Comment:
what do you think about using `path.posix.sep` instead of `/` ?

Reply:
I think it does not matter either way, changed